### PR TITLE
feat: Added support of group fields

### PR DESCRIPTION
**Pull request by tbalar-splunk**
JIRA Ticket: https://jira.splunk.com/browse/ADDON-35553

### CODE REVIEW

File: src/main/webapp/components/BaseFormView.jsx [hunk 1]
```diff
@@ -423,7 +514,11 @@ class BaseFormView extends PureComponent {
             >
                 {this.generateWarningMessage()}
                 {this.generateErrorMessage()}
+                {this.renderGroupElements()}
                 {this.entities.map((e) => {
+                    if (e.isGrouping) {
```
Comment (harshpatel-crest) [inline src/main/webapp/components/BaseFormView.jsx:519]: nit: Please add inline comment here mentioning why we need to handle this condition.

File: src/main/webapp/pages/style.scss [hunk 2]
```diff
@@ -1,8 +1,4 @@
-// @import '@splunk/themes/variables' as variable;
-@use '@splunk/themes/variables' as contentColorDefault;
-
 body {
-    // @include variables.contentColorDefault;
     min-width: 960px;
-    background-color: contentColorDefault;
+    background-color: #eee;
```
Comment (harshpatel-crest) [inline src/main/webapp/pages/style.scss:3]: Why are we adding it static?

File: src/main/webapp/components/BaseFormView.jsx [hunk 3]
```diff
@@ -48,7 +75,9 @@ class BaseFormView extends PureComponent {
         if (props.page === PAGE_INPUT) {
             globalConfig.pages.inputs.services.forEach((service) => {
                 if (service.name === props.serviceName) {
+                    this.groups = service.groups;
                     this.entities = service.entity;
+                    this.groupWiseEntities(service);
```
Comment (harshpatel-crest) [inline src/main/webapp/components/BaseFormView.jsx:80]: nit: This method is updating `this.entities` but is not clear from its name.

**Review by harshpatel-crest — APPROVED** [review]

File: src/main/webapp/components/BaseFormView.jsx [hunk 4]
```diff
@@ -13,6 +14,32 @@ import { axiosCallWrapper } from '../util/axiosCallWrapper';
 import TableContext from '../context/TableContext';
 import { parseErrorMsg } from '../util/messageUtil';
 
+const CollapsiblePanelWrapper = styled(CollapsiblePanel)`
```
Comment (mchavda-splunk) [inline src/main/webapp/components/BaseFormView.jsx:17]: Can we put this component in a separate file?

File: src/main/webapp/components/BaseFormView.jsx (hunk 3 above)
Comment (tbalar-splunk) [reply src/main/webapp/components/BaseFormView.jsx:80]: Done

File: src/main/webapp/components/BaseFormView.jsx (hunk 1 above)
Comment (tbalar-splunk) [reply src/main/webapp/components/BaseFormView.jsx:519]: Done

**Review by mamin-crest — APPROVED** [review]

File: src/main/webapp/components/BaseFormView.jsx (hunk 4 above)
Comment (tbalar-splunk) [reply src/main/webapp/components/BaseFormView.jsx:17]: @mchavda-splunk As discussed in a call, it is not required to create a new file for this styled component.
And this styled component is not used by any other component, hence for now it is not required to create a new file for this.

File: src/main/webapp/pages/style.scss (hunk 2 above)
Comment (tbalar-splunk) [reply src/main/webapp/pages/style.scss:3]: As discussed we are not sure about how to use Splunk UI's variable in a **.scss** file. Hence it's static. Once we get a solution to it we will change it later.

**Review by mchavda-splunk — APPROVED** [review]

**Review by dkhatri-crest — APPROVED** [review]

**Pull request merged**